lib/external/spokes.ts

```diff
@@ -0,0 +1,67 @@
+import {get, post} from "request-promise-native";
+import {LoggingHelper} from "../core/logging-helper";
+
+const Logger = "SPOKES-CLIENT";
+
+export class SpokesClient {
+    private _secretKey: string;
+    private _id: string;
+
+    public constructor(private id: string, private secretKey: string) {
+        this._secretKey = secretKey;
+        this._id = id;
+    }
+
+    public async verifyUUIDisNew(): Promise<boolean> {
+        const options = {
+            uri: `https://api.bespoken.link/pipe/${ this._secretKey }`,
+            headers: {
+                "x-access-token": "4772616365-46696f72656c6c61",
+            },
+            body: {
+
+            },
+            json: true // Automatically parses the JSON string in the response
+        };
+        try {
+            const response = await get(options);
+
+        } catch (error) {
+            if (error.statusCode && error.statusCode !== 404) {
+                // different kind of error, we log and throw
+                LoggingHelper.error(Logger, `Error while verifying id: ${ error.message }`);
+                throw error;
+            }
+            // uuid don't exist
+            return true;
+        }
+        return false;
```

Comment:
I didn't realize it was possible to just directly return a value for a promise - how does that work? Does typescript automatically wrap the boolean result in a promise?